onward/app/feed/DeeplyReadAgent.scala

```diff
@@ -75,26 +75,23 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
       slightly different states at any point in time, which is ok, as they converge at each refresh.
    */
 
-  private var ophanItems: Array[OphanDeeplyReadItem] = Array.empty[OphanDeeplyReadItem]
-
-  // Note that keys in pathToCapiContentMapping are paths without starting slash
-  private val pathToCapiContentMapping: scala.collection.mutable.Map[String, Content] =
-    scala.collection.mutable.Map.empty[String, Content]
+  private var deeplyReadItems: scala.collection.mutable.Map[OphanDeeplyReadItem, Content] =
+    scala.collection.mutable.Map.empty[OphanDeeplyReadItem, Content]
 
   def removeStartingSlash(path: String): String = {
     if (path.startsWith("/")) path.stripPrefix("/") else path
   }
 
-  def refresh()(implicit ec: ExecutionContext): Future[Unit] = {
+  def refresh()(implicit ec: ExecutionContext): Future[Seq[Option[(OphanDeeplyReadItem, Content)]]] = {
     log.info(s"[cb01a845] Deeply Read Agent refresh()")
     /*
         Here we simply go through the OphanDeeplyReadItem we got from Ophan and for each
         query CAPI and set the Content for the path.
      */
 
-    ophanApi.getDeeplyReadContent().map { seq =>
+    val deeplyReadFuture = ophanApi.getDeeplyReadContent().flatMap { seq =>
       log.info(s"[cb01a845] ophanItems updated with: ${seq.toArray.size} new items")
-      seq.foreach { ophanItem =>
+      val deeplyReadSeq = seq.map { ophanItem =>
```

Comment:
I think a part or all of this should be extracted into a separate function(s) (not necessarily right now), but calling it `seqCapiFutures` maybe helps for now.

Reply:
and putting the type on it